src/commons/workspace/WorkspaceReducer.ts

```diff
@@ -784,6 +786,34 @@ const newWorkspaceReducer = createReducer(defaultWorkspaceManager, builder => {
 
       state[workspaceLocation].editorTabs = newEditorTabs;
     })
+    .addCase(updateTabReadOnly, (state, action) => {
+      const workspaceLocation = getWorkspaceLocation(action);
+      const { editorTabIndex, isReadOnly } = action.payload;
+      if (isNull(editorTabIndex) || editorTabIndex < 0) {
+        return state;
+      }
+      const editorTabs = state[workspaceLocation].editorTabs;
+      const newEditorTabs = editorTabs.map((editorTab: EditorTabState, index: number | null) =>
+        index === editorTabIndex
+          ? {
+              ...editorTab,
+              readOnly: isReadOnly
+            }
+          : editorTab
+      );
+
+      if (isEqual(editorTabs, newEditorTabs)) {
```

Comment:
Equality does not work properly with drafts. See https://immerjs.github.io/immer/pitfalls/#drafts-arent-referentially-equal